sbi/inference/snpe/snpe_base.py

```diff
@@ -62,28 +73,21 @@ def __init__(
             show_round_summary=show_round_summary,
         )
 
+        # As detailed in the docstring, density estimator is either a string or
+        # a callable. The function creating the neural network is attached to
+        # `_build_neural_net`. It will be called in the first round and receive
+        # thetas and xs as inputs, so that they can be used for shape inference and
+        # potentially for z-scoring.
         if isinstance(density_estimator, str):
-            density_estimator = utils.posterior_nn(
-                model=density_estimator, prior=self._prior, x_o_shape=self._x_shape,
-            )
-
-        # Create a neural posterior which can sample(), log_prob().
-        self._posterior = NeuralPosterior(
-            method_family="snpe",
-            neural_net=density_estimator,
-            prior=self._prior,
-            x_shape=self._x_shape,
-            sample_with_mcmc=sample_with_mcmc,
-            mcmc_method=mcmc_method,
-            get_potential_function=PotentialFunctionProvider(),
-        )
+            self._build_neural_net = utils.posterior_nn(model=density_estimator)
```

Comment:
maybe pass the defaults for z_scoring etc here, otherwise they are hidden in the default of this utils function, no?

Reply:
I changed the default to 1e-14. I think it's a sane value and I can't really see scenarios where users would want to change it -- which is why I'm not exposing it to the inference classes. If you have a use-case/strong preference, let me know and I'll expose it